design/20220118.server-side-apply.md

```diff
@@ -132,36 +151,47 @@ All CREATE and UPDATE operations in cert-manager controllers are to be replaced
 with APPLY. For this change, controllers will need to be modified so that for
 these calls, _only_ the fields in which they manage should be included in the
 PATCH API call. This means each controller will need to define exactly which
-fields they are concerned with.
+fields they are concerned with. A package will be created which all API calls
+will use. This package will implement the logic for checking the feature gate
+for consistency across the project whilst minimising disruption to existing code
+paths.
 
 Some fields, such as the Certificate Issuing Condition are managed by more than
-one controller (issuing and trigger Certificate controllers), and as such, will
-need to make use of the `force` option in their API calls. This option tells the
-API server to revoke management of that field from the previous owner, overwrite
-the field, and change owner ship to the new client.
+one controller (issuing and trigger Certificate controllers, and cmctl), and as
+such, will need to make use of the `force` option in their API calls. This
```

Comment:
Thought: will there be any cases where `force` should _not_ be used?
I guess the assumption in this design is that _all_ conflicts that we get are caused by controllers modifying _different_ fields of the same resource (I assume that we cannot end up in a state where a controller has an older version of some resource (i.e due to some slow operations) and ends up force overwriting field X that was also modified by _another_ controller in a _newer_ version of the same resource).

Reply:
I will have a think, but I don't think there is any case where we do not want to use `force`. Getting a conflict error is an opportunity for the client to make a decision on whether they want to overwrite, co-own, or drop management of that value- all of our controllers own their managed fields and will want to overwrite.

Reply:
I'm a bit confused about using 'force' - this makes me think it is potentially overwriting other controller's progress.

Can we expand in the doc what force does, when it's appropriate to use, when we do use it, and why it's okay that we use it where we do/any special considerations?